src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,548 @@
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use vasi::VirtualAddressSpaceIndependent;
+
+use crate::bindings;
+
+pub const LINUX_STANDARD_SIGNAL_MAX_NO: i32 = 31;
+
+/// Lowest realtime signal number.
+pub const LINUX_SIGRT_MIN: i32 = bindings::SIGRTMIN as i32;
+/// Highest realtime signal number.
+//
+// According to signal(7). bindgen fails to bind this one.
+pub const LINUX_SIGRT_MAX: i32 = 64;
+
+/// Definition is sometimes missing in the userspace headers.
+//
+// bindgen fails to bind this one.
+// Copied from linux's include/uapi/linux/signal.h.
+pub const LINUX_SS_AUTODISARM: i32 = 1 << 31;
+
+// Bindgen doesn't succesfully bind these constants; maybe because
+// the macros defining them cast them to pointers.
+//
+// Copied from linux's include/uapi/asm-generic/signal-defs.h.
+pub const LINUX_SIG_DFL: usize = 0;
+pub const LINUX_SIG_IGN: usize = 1;
+pub const LINUX_SIG_ERR: usize = (-1_isize) as usize;
+
+// signal names
+#[derive(Debug, Copy, Clone, IntoPrimitive, TryFromPrimitive)]
+#[repr(i32)]
+pub enum LinuxSignal {
```

Comment:
I tried the `prefix` export option. The main place I ran into trouble is with exported functions from *other* crates. e.g. if we use `sigset_t` here, but use the `prefix` export option to export as `linux_sigset_t`, then in the `shadow-shim-helper-rs` crate when exporting APIs that take a pointer to this type, we have to account for the renaming somehow.

e.g. we now have a function in `shadow-shim-helper-rs` declared as:

```
    pub unsafe extern "C" fn shimshmem_getProcessPendingSignals(
        lock: *const ShimShmemHostLock,
        process: *const ShimShmemProcess,
    ) -> linux_sigset_t {
```

If we use `linux_api::sigset_t` instead of `linux_sigset_t`, then the cbindgen'd header will just call it `sigset_t` in the function signature. (Actually I'm not sure how this is working right in the code @ head, where it is `sigset_t`, and the shim code looks like it'd be getting the libc definition, which is the wrong size).

Ideally we'd want to configure cbindgen in the `shadow-shim-helper-rs` crate to prefix anything from the `linux-api` crate, but I don't think there's a way to do that. We'd have to enumerate the individual symbols to be renamed. If we forget one, it'll name the libc type instead, and might seem to work (but potentially fail in confusing ways at runtime).

I think I'd be happy to drop the `Linux` prefix on everything in this crate once we don't need to `cbindgen` it anymore :)